http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -423,6 +436,22 @@ public JettyHttpClient(
         });
     }
 
+    private static void setKeepAlive(SelectableChannel selectable, Duration tcpKeepAliveIdleTime)
+    {
+        if (selectable instanceof NetworkChannel channel) {
+            try {
+                channel.setOption(StandardSocketOptions.SO_KEEPALIVE, true);
+                channel.setOption(ExtendedSocketOptions.TCP_KEEPIDLE, toIntExact(tcpKeepAliveIdleTime.roundTo(SECONDS)));
+            }
+            catch (Throwable e) {
```

Comment:
Don’t carch Throwable as it can hide internal errors. Instead, catch the expected error.